src/alpaka/test/alpaka/OneHistoContainer_t.cc

```diff
@@ -62,83 +42,62 @@ struct mykernel {
     hist.finalize(acc, ws);
     alpaka::block::sync::syncBlockThreads(acc);
 
-    if (threadIdxLocal == 0) {
-      printf("hist.size() = %u.\n", hist.size());
-    }
     assert(N == hist.size());
 
     // verify
-    for (uint32_t threadIdx = firstElementIdxNoStride[0u], endElementIdx = endElementIdxNoStride[0u];
-         threadIdx < Hist::nbins();
-         threadIdx += blockDimension, endElementIdx += blockDimension) {
-      for (uint32_t j = threadIdx; j < std::min(endElementIdx, Hist::nbins()); ++j) {
-        assert(hist.off[j] <= hist.off[j + 1]);
-      }
-    }
+    cms::alpakatools::for_each_element_1D_block_stride(
+        acc, Hist::nbins(), [&](uint32_t j) { assert(hist.off[j] <= hist.off[j + 1]); });
     alpaka::block::sync::syncBlockThreads(acc);
 
-    if (threadIdxLocal < 32) {
-      ws[threadIdxLocal] = 0;  // used by prefix scan...
-    }
+    cms::alpakatools::for_each_element_in_thread_1D_index_in_block(acc, 32, [&](uint32_t i) {
+      ws[i] = 0;  // used by prefix scan...
+    });
```

Comment:
The old code did not have a loop over elements. I'm wondering which one is more correct.

Reply:
Both are fine:
The extra loop in Alpaka is due to the loop over elements (obviously not interesting for the GPU backend, but needs to be there for the sake of generality).
https://github.com/ghugo83/pixeltrack-standalone/blob/prefixscan_fix/src/alpaka/AlpakaCore/alpakaWorkDivHelper.h#L163
Whenever the thread index (block-local) is >= 32, the thread index is still called and compared with 32 (like in legacy). But the loop over elements is never entered (and no ws[i]=0 allocation is done), because in that case, `firstElementIdx[0u] > endElementIdx[0u]`